pkg/sysinfo/sysinfo.go

```diff
@@ -102,6 +117,128 @@ type cgroupPids struct {
 	PidsLimit bool
 }
 
+// ValidateHugetlb check whether hugetlb pagesize and limit legal
+func (c cgroupHugetlbInfo) ValidateHugetlb(pageSize string, limit uint64) (string, []string, error) {
+	var (
+		w   []string
+		err error
+	)
+	if pageSize != "" {
+		if err = isHugepageSizeValid(pageSize); err != nil {
+			return "", w, err
+		}
+	} else {
+		pageSize, err = c.GetDefaultHugepageSize()
+		if err != nil {
+			return "", w, fmt.Errorf("Failed to get system hugepage size")
+		}
+	}
+
+	warning, err := isHugeLimitValid(pageSize, limit)
+	w = append(w, warning...)
+	if err != nil {
+		return "", w, err
+	}
+
+	return pageSize, w, nil
+}
+
+// isHugeLimitValid check whether input hugetlb limit legal
+// it will check whether the limit size is times of size
+func isHugeLimitValid(size string, limit uint64) ([]string, error) {
+	var w []string
+	sizeInt, err := units.RAMInBytes(size)
+	if err != nil || sizeInt < 0 {
+		return w, fmt.Errorf("Invalid hugepage size:%s -- %s", size, err)
+	}
+	sizeUint := uint64(sizeInt)
+
+	if limit%sizeUint != 0 {
+		w = append(w, "Invalid hugetlb limit: should be times of huge page size")
+	}
+
+	return w, nil
+}
+
+// isHugepageSizeValid check whether input size legal
+// it will compare size with all system supported hugepage size
+func isHugepageSizeValid(size string) error {
+	hps, err := getHugepageSizes()
+	if err != nil {
+		return err
+	}
+
+	for _, hp := range hps {
+		if size == hp {
+			return nil
+		}
+	}
+	return fmt.Errorf("Invalid hugepage size:%s, shoud be one of %v", size, hps)
+}
+
+func humanSize(i int64) string {
```

Comment:
If we need some specific conversion, then we should either do it in the units package , or directly where it's being read (sysinfo I guess here?).

However conversions from user input really should be consistent.